src/compiler/types.ts

```diff
@@ -1606,6 +1606,16 @@ namespace ts {
         antecedent: FlowNode;
     }
 
+    export type FlowType = Type | IncompleteType;
+
+    // Incomplete types occur during control flow analysis of loops. An IncompleteType
+    // is distinguished from a regular type by a flags value of zero. Incomplete type
+    // objects are internal to the getFlowTypeOfRefecence function and never escape it.
+    export interface IncompleteType {
+        flags: TypeFlags;  // No flags set

```

Comment:
Since we have numeric literal types and enum literal types now, shouldn't this just be `TypeFlags.None` or `0`? (Which would alleviate the need for the long explanatory comment above.)